src/ray/gcs/redis_client.cc

```diff
@@ -132,6 +132,12 @@ Status RedisClient::Connect(std::vector<boost::asio::io_service *> io_services)
       boost::asio::io_service &io_service = *io_services[io_service_index];
       // Populate shard_contexts.
       shard_contexts_.push_back(std::make_shared<RedisContext>(io_service));
+      Status portReachable = shard_contexts_[i]->PingPort(addresses[i], ports[i]);
+      if (!portReachable.ok()) {
+        RAY_LOG(WARNING) << "Successfully connected to " << options_.server_ip_ << ":"
+                         << options_.server_port_ << ", but not to the shard address it "
+                         << "sent: " << portReachable.message();
+      }
```

Comment:
thanks for this PR, @dHannasch -- I merged in 12026 for you.

Can you actually help provide an example for where this is useful?